terraform/modules/frontdoor/load_balancer.tf

```diff
@@ -0,0 +1,62 @@
+#tfsec:ignore:aws-elb-alb-not-public:the load balancer must be exposed to the internet in order to communicate with cloudfront
+resource "aws_lb" "main" {
+  name                       = "alb-${var.environment_name}"
+  drop_invalid_header_fields = true
+  enable_deletion_protection = true
+  internal                   = false
+  load_balancer_type         = "application"
+  security_groups            = [aws_security_group.load_balancer.id]
+  subnets                    = var.public_subnet_ids
+}
+
+resource "aws_lb_listener" "https" {
+  certificate_arn   = var.load_balancer_certificate_arn
```

Comment:
This means we'll need to create new environments in two steps - one to create and validate the certificates, and a second to create the rest of the infra. This seems to be a relatively common pattern and I couldn't see a way around it.

Reply:
Yup this seems fine to me, as long as it's documented in the deployment guide for a new environment :)